bin/metric-zfs-arc.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+#
+#   ZFS ARC metrics
+#
+# DESCRIPTION:
+#   This collects ZFS ARC (Adaptive Replacement Cache) metrics
+#   metric-zfs-arc.rb looks at /proc/spl/kstat/zfs/arcstats
+#
+# OUTPUT:
+#  metric data
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: socket
+#
+# USAGE:
+#  ./metric-zfs-arc.rb
+#
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2019 Airbrake Technologies, Inc <support@airbrake.io>
+#   Released under the same terms as the Sensu (the MIT license); see LICENSE
+#   for details.
+
+require 'sensu-plugin/metric/cli'
+require 'socket'
+
+#
+# ZFS Arc Metrics
+##
+class ZfsArcMetrics < Sensu::Plugin::Metric::CLI::Graphite
+  option :scheme,
+         description: 'Metric naming scheme, text to prepend to .$parent.#child',
+         short: '-s SCHEME',
+         long: '--scheme SCHEME',
+         default: "#{Socket.gethostname}.zfs"
+
+  def run
+    if File.exist?('/proc/spl/kstat/zfs/arcstats')
+      command = 'cat /proc/spl/kstat/zfs/arcstats'
+
+      `#{command}`.split("\n").drop(2).each do |row|
```

Comment:
we are executing the results from `/proc/spl/kstat/zfs/arcstats` sorry I don't run ZFS anywhere myself can you expand on what this file contains?

Reply:
ZFS reports information about ARC (Adaptive Replacement Cache) performance to `/proc/spl/kstat/zfs/arcstats`. It contains metrics to help determine how your RAM is being used and whether / where you might need to make changes improve ZF S performance.

Reply:
I just realized were not executing the results, that was simply a command string. It makes more sense. With systems that are more "exotic" or less common I encourage adding a comment above parsing lines like these with a sample of what the line looks like. This helps us out when people report bugs we can notice say changes to the output format without having to setup the specific tech itself. Not a requirement but something I highly recommend doing. This one is fairly easy but when say a complex regex is used it really does help.

Reply:
Added